spices/SPICE-0001-import-graph-analyzer-api.adoc

```diff
@@ -0,0 +1,336 @@
+import* "path/to/other/*.pkl" as otherModules // <3>
+
+res1 = import("some/other/module.pkl") // <4>
+
+res2 = import*("some/other/*.pkl") // <5>
+----
+<1> Amends clause
+<2> Import clause
+<3> Globbed import clause
+<4> Import expression
+<5> Globbed import expression
+
+=== CLI Command: `pkl analyze imports`
+
+A new command will be added to the CLI, called `pkl analyze imports`.
+
+Syntax:
+
+[source]
+----
+pkl analyze imports [-f,--format <format>] [--project-dir <path>] <moduleUri> [...moduleUri]
+----
+
+This command parses the provided source modules, and resolves all imported modules to their absolute URI.
+
+By default, the result is Pcf. Example:
+
+[source,shell]
+----
+pkl analyze imports myFile.pkl
+----
+
+Produces:
+[source]
+----
+imports {
+  ["file:///path/to/myFile.pkl"] {
+    "file:///path/to/otherFile.pkl"
+  }
+  ["file:///path/to/otherFile.pkl"] {}
+}
+resolvedUris {
+  ["file:///path/to/myFile.pkl"] = "file:///path/to/myFile.pkl"
+  ["file:///path/to/myFile.pkl"] = "file:///path/to/myFile.pkl"
+}
+----
+
+The value of the `imports` property is an object whose keys are module URIs, and whose values are the set of modules imported by the module referenced by the key.
+
+The set of all input modules in a program is the set of keys in the `imports` object.
+
+The value of `resolvedUris` is the mapping of in-language module URIs to those that Pkl uses to determine how to fetch the module's contents.
+For example, in the case of a link:https://pkl-lang.org/main/current/language-reference/index.html#local-dependencies[local dependency], an in-language URI might look like `projectpackage://example.com/foo@1.0.0#/mod.pkl`, and its resolved URI might look like `\file:///path/to/foo/mod.pkl`.
+
+The resolved URIs help build systems understand how resources map
+
+Underneath the hood, the command builds a synthetic module, that ultimately uses standard library module <<stdlib-module-pkl-analyze,pkl.analyze>> to produce output.
+
+==== `--format`, `-f`
+
+The `--format` (`-f` for short) flag sets the output format for this command.
+
+Examples:
+
+[source,shellscript]
+----
+# Evaluate as JSON
+$ pkl analyze imports -f json myFile.pkl
+
+# Evaluate as YAML
+$ pkl analyze imports -f yaml myFile.pkl
+----
+
+==== Common CLI flags
+
+This command recursively traverses the import graph.
+To support this, the set of flags from common evaluator settings are available:
+
+* `--allowed-modules`
+* `--allowed-resources`
+* `--cache-dir`
+* `--module-path`
+* `--project-dir`
+* `--ca-certificates`
+
+[[stdlib-module-pkl-analyze]]
+=== Standard library module `pkl:analyze`
+
+A new standard library module is introduced.
+It contains the class `ImportAnalyzer`, which provides import graph analysis.
+
+[source,pkl]
+----
+/// Utilities for static analysis of Pkl modules.
+///
+/// Unlike `pkl:reflect`, these methods do not require loading the modules for analysis.
+@ModuleInfo { minPklVersion = "0.26.0" }
+@Since { version = "0.26.0" }
```

Comment:
I know it's just an example, but still better to change this and `minPklVersion` to `0.27.0` to be correct.